django/thunderstore/api/cyberstorm/views/package_listing_actions.py

```diff
@@ -123,3 +128,53 @@ def post(self, request, *args, **kwargs) -> Response:
         )
 
         return Response({"message": "Success"}, status=status.HTTP_200_OK)
+
+
+class ReportPackageListingAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    @conditional_swagger_auto_schema(
+        operation_id="cyberstorm.package_listing.report",
+        tags=["cyberstorm"],
+        request_body=PackageListingReportRequestSerializer,
+        responses={200: "Success"},
+    )
+    def post(self, request, *args, **kwargs) -> Response:
+        listing: PackageListing = get_package_listing(
+            namespace_id=kwargs["namespace_id"],
+            package_name=kwargs["package_name"],
+            community_id=kwargs["community_id"],
+        )
+
+        request_serializer = PackageListingReportRequestSerializer(data=request.data)
+        request_serializer.is_valid(raise_exception=True)
+
+        report_package_listing(
+            agent=request.user,
+            reason=request_serializer.validated_data.get("reason"),
+            package=listing.package,
+            package_listing=listing,
```

Comment:
I feel like these could be consolidated to just taking `package_listing` as input? Package is always linked to a listing after all